src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java

```diff
@@ -0,0 +1,207 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with Sonar; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package hudson.plugins.sonar;
+
+import org.apache.commons.lang.StringUtils;
+
+import hudson.plugins.sonar.utils.MaskPasswordsOutputStream;
+import hudson.EnvVars;
+import hudson.plugins.sonar.utils.Logger;
+
+import javax.annotation.Nullable;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import hudson.CopyOnWrite;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.BuildListener;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SonarBuildWrapper extends BuildWrapper {
+  private String installationName = null;
+
+  @DataBoundConstructor
+  public SonarBuildWrapper(@Nullable String installationName) {
+    this.installationName = installationName;
+  }
+
+  @Override
+  public DescriptorImpl getDescriptor() {
+    return (DescriptorImpl) super.getDescriptor();
+  }
+
+  @Override
+  public OutputStream decorateLogger(AbstractBuild build, OutputStream outputStream) throws IOException, InterruptedException {
+    SonarInstallation inst = getInstallation();
+    if (inst == null || inst.isDisabled() || (inst.getDatabasePassword() == null && inst.getSonarPassword() == null)) {
+      return outputStream;
+    }
+
+    Logger.LOG.info(Messages.SonarBuildWrapper_MaskingPasswords());
+
+    List<String> passwords = new ArrayList<String>();
+
+    if (inst.getDatabasePassword() != null) {
+      passwords.add(inst.getDatabasePassword());
+    }
+    if (inst.getSonarPassword() != null) {
+      passwords.add(inst.getSonarPassword());
+    }
+
+    return new MaskPasswordsOutputStream(outputStream, passwords);
+  }
+
+  /**
+   * @return name of {@link hudson.plugins.sonar.SonarInstallation}
+   */
+  @Nullable
+  public String getInstallationName() {
+    return installationName;
+  }
+
+  public void setInstallationName(@Nullable String installationName) {
+    this.installationName = installationName;
+  }
+
+  @Nullable
+  public SonarInstallation getInstallation() {
+    return SonarInstallation.get(getInstallationName());
+  }
+
+  @Override
+  public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
+    if (!SonarRunnerBuilder.isSonarInstallationValid(getInstallationName(), listener)) {
+      return new Environment() {
+      };
+    }
+    return new SonarEnvironment(getInstallation(), listener.getLogger());
+  }
+
+  @Extension
+  public static final class DescriptorImpl extends BuildWrapperDescriptor
+  {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move this left curly brace to the end of previous line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3ALeftCurlyBraceEndLineCheck)